src/Compilers/Core/Desktop/AnalyzerFileReference.cs

```diff
@@ -208,13 +213,12 @@ internal void AddAnalyzers(ImmutableArray<DiagnosticAnalyzer>.Builder builder, s
                 // If there are no analyzers, don't load the assembly at all.
                 if (!analyzerTypeNameMap.ContainsKey(language))
                 {
-                    this.AnalyzerLoadFailed?.Invoke(this, new AnalyzerLoadFailureEventArgs(AnalyzerLoadFailureEventArgs.FailureErrorCode.NoAnalyzers, null, null));
```

Comment:
@mavasani Wouldn't the error you're describing arise during the call to `GetLanguageSpecificAnalyzers` (line 231)? `GetAnalyzerTypeNameMap` is just looking for types with an attribute of the proper name using the metadata reader; whether or not the type can actually be loaded/used is a different matter.

So we will still raise an error if we find something claiming to be an analyzer (that is. it has the attribute) but can't be loaded for whatever reason.